### PR TITLE
sensor: add sensor driver for DPS310 and example

### PR DESCRIPTION
**Pull request by creiter32**
Add driver for Infineon DPS310 temperature and pressure sensor.

Signed-off-by: Christoph Reiter <christoph.reiter@infineon.com>

### CODE REVIEW

File: samples/sensor/dps310/README.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+.. _dps310:
+
+DPS310 Temperature and Pressure Sensor
+###################################
```
Comment (dbkinder) [inline samples/sensor/dps310/README.rst]: Title underline needs to be at least as long as the title, so add some more ``###``s

File: samples/sensor/dps310/README.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+interrupt trigger).
+
+Building and Running
+********************
+
+This sample application uses an DPS310 sensor connected to a board via I2C.
+Connect the sensor pins according to the connection diagram given in the `dps310 datasheet`_
+at page 18 figure 7.
+
+
```
Comment (dbkinder) [inline samples/sensor/dps310/README.rst]: Missing an intro sentence here:
```
Build and flash this sample (for example, for the nrf52840_pca10056 board) using these commands:

.. zephyr-app-commands::
```

File: samples/sensor/dps310/README.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+.. zephyr-app-commands::
+   :zephyr-app: samples/sensors/dps310
+   :board: nrf52840_pca10056
+   :goals: flash
+   :compact:
+
+Sample Output
+=============
+To check output of this sample , any serial console program can be used.
```
Comment (dbkinder) [inline samples/sensor/dps310/README.rst]: Delete the space before the comma.

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by creiter32** [comment]
I additionally limited the line lenght to 80 characters where possible

**Review by dbkinder — APPROVED** [review]
+1 for doc changes, thanks.

**Comment by creiter32** [comment]
CC: @carlescufi as requested Mihai

File: drivers/sensor/dps310/dps310.c [hunk 4]
```diff
@@ -0,0 +1,760 @@
...
+	const struct dps310_cfg *config = dev->config->config_info;
+
+	data->i2c_master = device_get_binding(config->i2c_bus_name);
+	if (data->i2c_master == NULL) {
+		LOG_ERR("Failed to get I2C device");
+		return -EINVAL;
+	}
+
+	u8_t product_id = 0;
+	int res = i2c_write_read(data->i2c_master, config->i2c_addr,
```
Comment (MaureenHelm) [inline drivers/sensor/dps310/dps310.c]: Use `i2c_reg_read_byte`

File: samples/sensor/dps310/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
```
Comment (MaureenHelm) [inline samples/sensor/dps310/CMakeLists.txt]: Please add the sample in a separate commit in the same PR.

File: samples/sensor/dps310/sample.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+sample:
+  name: DP310 Sensor sample
+tests:
+  sample.sensor.dps310:
+    harness: console
+    tags: sensors
+    depends_on: i2c dps310
```
Comment (MaureenHelm) [inline samples/sensor/dps310/sample.yaml]: Unless you add `dps310` to the `supported` list in a board, this sample will never build in CI.

File: samples/sensor/dps310/src/main.c [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/sensor.h>
+#include <stdlib.h>
+
+void main(void)
+{
+	printk("Hello DPS310\n");
+	struct device *dev = device_get_binding("DPS310");
```
Comment (MaureenHelm) [inline samples/sensor/dps310/src/main.c]: Use `DT_INST_0_INFINEON_DPS310_LABEL` instead

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please add this sensor to `tests/drivers/build_all/sensors_a_h.conf`

File: drivers/sensor/dps310/Kconfig [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+#
+# Copyright (c) 2019 Infineon Technologies AG
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+menuconfig DPS310
+	bool "DPS310 sensor"
+	depends on I2C
+	default	n
```
[the remark this answers is not in the document]
Comment (creiter32) [reply drivers/sensor/dps310/Kconfig]: Done

File: drivers/sensor/dps310/dps310.c [hunk 9]
```diff
@@ -0,0 +1,760 @@
...
+	u16_t i2c_addr;
+};
+
+/* Convert two bytes from the calibration memory to a s16_t value */
+static s16_t raw_calib_to_s16(const u8_t raw[2])
+{
+	/* values are store in little endian on the chip */
+	s32_t tmp = raw[1] + (((u16_t)raw[0]) << 8);
+	/* make value negative by converting from 2's complement */
+	tmp = tmp > POW_2_15_MINUS_1 ? tmp - POW_2_16 : tmp;
```
[the remark this answers is not in the document]
Comment (creiter32) [reply drivers/sensor/dps310/dps310.c]: Done

File: drivers/sensor/dps310/dps310.c [hunk 10]
```diff
@@ -0,0 +1,760 @@
...
+	/* convert raw data to int */
+	data->raw_tmp = raw_to_int24(&value_raw[0]);
+
+	dps310_scale_temperature(data->raw_tmp, data);
+
+	return true;
+}
+
+/* Initialize the sensor and apply the configuration */
+int dps310_init(struct device *dev)
```
[the remark this answers is not in the document]
Comment (creiter32) [reply drivers/sensor/dps310/dps310.c]: Done

File: samples/sensor/dps310/CMakeLists.txt (hunk 5 above)
Comment (creiter32) [reply samples/sensor/dps310/CMakeLists.txt]: Done

File: samples/sensor/dps310/src/main.c (hunk 7 above)
Comment (creiter32) [reply samples/sensor/dps310/src/main.c]: Done

File: samples/sensor/dps310/sample.yaml (hunk 6 above)
Comment (creiter32) [reply samples/sensor/dps310/sample.yaml]: Where do I have to add it? The sample should run on any board that has the Arduino I2C interface defined.

Comment (carlescufi) [reply samples/sensor/dps310/sample.yaml]: I guess it's arbitrary in this case. Do we have a rule for this @MaureenHelm, @galak, @pabigot ?

Comment (pabigot) [reply samples/sensor/dps310/sample.yaml]: Maybe `depends_on: i2c arduino_i2c`?

Comment (creiter32) [reply samples/sensor/dps310/sample.yaml]: Changed to `depends_on: i2c arduino_i2c`

**Comment by creiter32** [comment]
It seems to me that the CI is failing because the DPS310 sensor is not in the device tree of the test boards.

How can I tell the test system to use the `dps310.dts` overlay?

**Comment by creiter32** [comment]
ping @MaureenHelm @galak @nashif

**Comment by creiter32** [comment]
Hi, is there no interest to merge this driver?

**Pull request closed without merging**